### PR TITLE
Hypothesis tests for VCF

### LINKED ISSUE #249: Hypothesis testing for vcf2zarr
It would be good to use the [Hypothesis strategy for generating VCF files](https://github.com/sgkit-dev/sgkit/blob/27d77799d3e42a01ef263108f1cabcd33735986a/sgkit/tests/io/vcf/hypothesis_vcf.py#L232-L344) that's in sgkit against vcf2zarr, to check for corner cases in conversion.

I wonder if we should move the Hypothesis VCF code to this repo, or release as a separate package (it may be of general interest)?

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #249

This should be working now, but does it need a hypothesis-vcf release before merging (or not)?

### CODE REVIEW

**Comment by tomwhite** [comment]
The failure here was because hypothesis created a VCF with a GT INFO field of type Character, and bio2zarr assumes all GT fields have type Integer. The VCF is valid (but almost certainly never occurs in practice), so I think the fix is to change bio2zarr to only interpret GT FORMAT fields as Integer.

**Review by jeromekelleher — APPROVED** [review]
LGTM!

I don't really have an opinion about making a release of hypothesis VCF, probably doesn't matter much when it's a dev dependency?

File: tests/test_vcf_hypothesis.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import pysam
+from hypothesis import HealthCheck, given, note, settings
+from hypothesis_vcf import vcf
+
+from bio2zarr import vcf2zarr
+
+# Make sure POS starts at 1, since CSI indexing doesn't seem to support zero-based coordinates
+# (even when passing zerobased=True to pysam.tabix_index below)
+@given(vcf_string=vcf(min_pos=1))
```
[the remark this answers is not in the document]
Comment (jeromekelleher) [reply tests/test_vcf_hypothesis.py:12]: Aha, I see. Very good.

**Comment by jeromekelleher** [comment]
I'm happy to merge if you are?
